### PR TITLE
add com.w1hkj.flcluster

### PR DESCRIPTION
**Pull request by lu9dce**
flcluster is a stand alone dx cluster client that will also connect to fldigi
and act as the dx cluster client for fldigi. Multiple instances of flcluster   
can simultaneously connect to fldigi, and fldigi's internal dx cluster client
can also be connected at the same time.  Connectivity is via a socket interface,
so flclient can reside on the same or a different computer on the LAN or WAN
so long as it is possible to establish the socket (xmlrpc) connection.

### CODE REVIEW

**Comment by lu9dce** [comment]
@bilelmoussaoui all running and testing ... send the bot tks

**Comment by lu9dce** [comment]
bot, build com.w1hkj.flcluster

**Comment by lu9dce** [comment]
bot, build com.w1hkj.flcluster

**Comment by lu9dce** [comment]
bot, build com.w1hkj.flcluster

File: com.w1hkj.flcluster.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+app-id: com.w1hkj.flcluster
+runtime: org.freedesktop.Platform
+runtime-version: '20.08'
+sdk: org.freedesktop.Sdk
+command: flcluster
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
```
Comment (barthalion) [inline com.w1hkj.flcluster.yml:9]:
```suggestion
  - --socket=x11
```

From what I know, FLTK does not support Wayland.

File: com.w1hkj.flcluster.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  - /doc
+  - '*.a'
+  - '*.la'
+modules:
+  - name: fltk
+    sources:
+      - type: archive
+        url: https://www.fltk.org/pub/fltk/1.3.5/fltk-1.3.5-source.tar.bz2
+        sha256: 2933c72400f9e7c0f4c3a81a1ce98bc9582b2a3edc44b8597ccd26e240e32c3c
+  - name: fldigi
```
Comment (barthalion) [inline com.w1hkj.flcluster.yml:27]:
```suggestion
  - name: flcluster
```

File: com.w1hkj.flcluster.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  so long as it is possible to establish the socket (xmlrpc) connection.
+  </p>
+ </description>
+ <screenshots>
+  <screenshot type="default">
+   <image>http://www.w1hkj.com/flcluster-help/flcluster.png</image>
+  </screenshot>
+ </screenshots>
+ <releases>
+   <release version="1.0.4" date="2020-11-1"/>
```
Comment (barthalion) [inline com.w1hkj.flcluster.metainfo.xml:27]:
```suggestion
   <release version="1.0.4" date="2020-11-01"/>
```

**Comment by lu9dce** [comment]
bot, build com.w1hkj.flcluster

**Comment by lu9dce** [comment]
bot, build com.w1hkj.flcluster

**Comment by lu9dce** [comment]
Well we can say that flcluster that well
what would be the step to finish the job?

flcluster

File: com.w1hkj.flcluster.yml [hunk 4]
```diff
@@ -9,6 +9,9 @@ finish-args:
   - --device=all
   - --share=network
   - --socket=pulseaudio
+  - --persist=.flcluster
+  - --persist=.fltk
+  - --persist=.nbems
```
Comment (AsciiWolf) [inline com.w1hkj.flcluster.yml:14]: The `.nbems` directory does not seem to be used by flcluster. You can probably remove it.

**Comment by AsciiWolf** [comment]
Hmm, not sure what happened but flcluster crashed when I tested it (connected to the dxfun.com cluster):
```

/usr/include/c++/10.2.0/bits/basic_string.h:1038: std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::reference std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::operator[](std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::reference = char&; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]: Assertion '__pos <= size()' failed.
Aborted (core dumped)
```

**Comment by lu9dce** [comment]
> Hmm, not sure what happened but flcluster crashed when I tested it (connected to the dxfun.com cluster):
> 
> ```
> 
> /usr/include/c++/10.2.0/bits/basic_string.h:1038: std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::reference std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::operator[](std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::reference = char&; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]: Assertion '__pos <= size()' failed.
> Aborted (core dumped)
> ```

The only distro that I use on all my machines is slackware 14.2 .. and the flcluster works fine
although this should not be affected by flatpak
Captura de pantalla_2020-12-01_10-02-23

**Comment by lu9dce** [comment]
test cluster iw9fdd.homeip.net 7300

**Comment by lu9dce** [comment]
**### ohhhhhh**

/usr/include/c++/10.2.0/bits/basic_string.h:1038: std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::reference std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::operator[](std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type) [with _CharT = char; _Traits = std::char_traits<char>; _Alloc = std::allocator<char>; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::reference = char&; std::__cxx11::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]: Assertion '__pos <= size()' failed.

**Comment by lu9dce** [comment]
org.freedesktop.Sdk 18.08.39 ???

**Comment by AsciiWolf** [comment]
The 18.08 Sdk/Runtime is too old.

File: com.w1hkj.flcluster.yml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+command: flcluster
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --device=all
+  - --share=network
+  - --socket=pulseaudio
+  - --persist=.flcluster
+  - --persist=.fltk
+  - --persist=.nbems
```
Comment (AsciiWolf) [inline com.w1hkj.flcluster.yml:14]: Yeah, it really looks like that this dir is not used by flcluster. Feel free to remove it.

**Comment by lu9dce** [comment]
looking at compilation both fldigi and flcluster show c ++ 10 warning
I'm going to try Sdk 18.08
Keep in mind that radio programs are not very modern .. for example in slack 14.2 they run perfectly directly but with fedora 33 I have several problems, mainly with the qt5 version

**Comment by lu9dce** [comment]
**_### in Sdk 18.08 no close !!! O.o_**

**Comment by lu9dce** [comment]
@AsciiWolf  if you enlarge the size of the window it does not close
now if you leave it at the default size
produces the error
can you try this?
maximizes or enlarges the size of the flcluster window

**Comment by lu9dce** [comment]
@AsciiWolf 
if you enlarge the size of the window it does not close
now if you leave it at the default size
produces the error
can you try this?
maximizes or enlarges the size of the flcluster window

**Comment by lu9dce** [comment]
_Unfortunately for some reason it closes, I will try another version of flcluster_

File: com.w1hkj.flcluster.yml (hunk 4 above)
Comment (AsciiWolf) [reply com.w1hkj.flcluster.yml:14]: Yeah, it really looks like that this dir is not used by flcluster. @lu9dce Feel free to remove it.

**Comment by AsciiWolf** [comment]
Another possible "solution" would be to close this PR. QLog seems to have integrated basic DX cluster support.

**Comment by lu9dce** [comment]
ok .. yes .. I tried several versions and it ends up closing ..
well I close the pr
thank you teacher for your help

**Pull request closed without merging**